test/spectests/spectests.cpp

```diff
@@ -122,25 +123,18 @@ class test_runner
 
                     const auto expected_type = expected.at(0).at("type").get<std::string>();
                     uint64_t expected_value;
-                    uint64_t actual_value;
                     if (expected_type == "i32")
-                    {
                         expected_value = json_to_value<int32_t>(expected.at(0).at("value"));
-                        actual_value =
-                            static_cast<uint64_t>(static_cast<int32_t>(result->stack[0]));
-                    }
                     else if (expected_type == "i64")
-                    {
                         expected_value = json_to_value<int64_t>(expected.at(0).at("value"));
-                        actual_value = result->stack[0];
-                    }
                     else
                     {
                         skip("Unsupported expected type '" + expected_type + "'.");
                         continue;
                     }
 
-                    if (expected_value != actual_value)
+                    const uint64_t actual_value = result->stack[0];
```

Comment:
We don't cast actual value anymore, bacause if it's 32-bit, returned stack value must have 0s in the high 32 bits.